hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/test/GenericTestUtils.java

```diff
@@ -345,6 +356,33 @@ public static void assertExceptionContains(String expectedText,
     }
   }
 
+  /**
+   * Assert that an exception's <code>toString()</code> value
+   * matches the pattern.
+   * @param pattern regex pattern to match
+   * @param t thrown exception
+   * @param message any extra text for the string
+   * @throws AssertionError if the expected string is not found
+   */
+  public static void assertExceptionMatches(Pattern pattern,
+      Throwable t,
+      String message) {
+    assertNotNull(t, E_NULL_THROWABLE);
+    String msg = t.toString();
+    if (msg == null) {
+      throw new AssertionError(E_NULL_THROWABLE_STRING, t);
+    }
+    if (pattern != null && !pattern.matcher(msg).matches()) {
+      String prefix = org.apache.commons.lang3.StringUtils.isEmpty(message)
```

Comment:
our own StringUtils.hasLength(message) does this. silly name but at least unique.